biz.aQute.bndlib/src/aQute/bnd/http/HttpClient.java

```diff
@@ -78,7 +78,7 @@ synchronized void init() {
 			return;
 
 		inited = true;
-		progress = registry != null ? registry.getPlugin(ProgressPlugin.class) : null;
+		progressPlugins = registry != null ? registry.getPlugins(ProgressPlugin.class) : null;
 
```

Comment:
I am not sure it is a good idea to just collect the list of ProgressPlugins in init since the list can change after that.